xbmc/pvr/providers/PVRProviders.cpp

```diff
@@ -377,11 +377,12 @@ void CPVRProviders::RemoveEntry(const std::shared_ptr<CPVRProvider>& provider)
 {
   CSingleLock lock(m_critSection);
 
-  std::remove_if(m_providers.begin(), m_providers.end(),
-                 [&provider](const std::shared_ptr<CPVRProvider>& providerToRemove) {
-                   return provider->GetClientId() == providerToRemove->GetClientId() &&
-                          provider->GetUniqueId() == providerToRemove->GetUniqueId();
-                 });
+  m_providers.erase(
+      std::remove_if(m_providers.begin(), m_providers.end(),
+                     [&provider](const std::shared_ptr<CPVRProvider>& providerToRemove) {
+                       return provider->GetClientId() == providerToRemove->GetClientId() &&
+                              provider->GetUniqueId() == providerToRemove->GetUniqueId();
+                     }));
```

Comment:
I think you need to provide an end also.

```suggestion
                     }), m_providers.end());
```

Reply:
For reference: https://godbolt.org/z/3Te7xejdc

Reply:
right, thanks